ts/features/bonus/siciliaVola/saga/orchestration/voucherGeneration.ts

```diff
@@ -0,0 +1,39 @@
+import { call } from "redux-saga/effects";
+import { SagaIterator } from "redux-saga";
+import {
+  executeWorkUnit,
+  withResetNavigationStack
+} from "../../../../../sagas/workUnit";
+import {
+  svGenerateVoucherBack,
+  svGenerateVoucherCancel,
+  svGenerateVoucherCompleted,
+  svGenerateVoucherFailure
+} from "../../store/actions/voucherGeneration";
+import SV_ROUTES from "../../navigation/routes";
+import { navigateToSvCheckStatusRouterScreen } from "../../navigation/actions";
+
+/**
+ * Define the workflow that allows the user to generate a new voucher.
+ * The workflow ends when:
+ * - The user send the request to generate a new voucher {@link svGenerateVoucherCompleted}
+ * - The user aborts the voucher generation {@link svGenerateVoucherCancel}
+ * - The user chooses back from the first screen {@link svGenerateVoucherBack}
+ */
+function* svVoucherGenerationWorkUnit() {
+  return yield call(executeWorkUnit, {
+    startScreenNavigation: navigateToSvCheckStatusRouterScreen(),
+    startScreenName: SV_ROUTES.VOUCHER_GENERATION.CHECK_STATUS,
+    complete: svGenerateVoucherCompleted,
+    back: svGenerateVoucherBack,
+    cancel: svGenerateVoucherCancel,
+    failure: svGenerateVoucherFailure
+  });
+}
+
+/**
+ * This saga handles the SV activation workflow
+ */
+export function* handleSvVoucherGenerationStartActivationSaga(): SagaIterator {
+  yield call(withResetNavigationStack, svVoucherGenerationWorkUnit);
```

Comment:
```suggestion
  const sagaExecution = () =>
    withFailureHandling(() =>
      withResetNavigationStack(svVoucherGenerationWorkUnit)
    );
  yield call(sagaExecution);
```
If you want the automatic handling for failures with the generic screen. If you intend to custom handling the failures in a next implementation, please ignore this suggestion!

Reply:
You're right. Updated in [4ddd528](https://github.com/pagopa/io-app/pull/3128/commits/4ddd5283d71ea7f9d8e6603ee4b7feac0cce3981)